File: java/micronaut-postgres/app/src/main/java/com/example/service/TelemetryService.java
package com.example.service;

import io.opentelemetry.api.GlobalOpenTelemetry;
import io.opentelemetry.api.metrics.LongCounter;
import io.opentelemetry.api.metrics.Meter;
import jakarta.annotation.PostConstruct;
import jakarta.inject.Singleton;

@Singleton
public class TelemetryService {

    private LongCounter articlesCreated;

    @PostConstruct
    void init() {
        Meter meter = GlobalOpenTelemetry.getMeter("micronaut-articles");
        articlesCreated = meter.counterBuilder("articles.created")
                .setDescription("Total number of articles created")
                .build();
    }

    public void incrementArticlesCreated() {
        articlesCreated.add(1);
    }
}

File: kotlin/ktor-postgres/app/src/main/kotlin/com/example/service/TelemetryService.kt
package com.example.service

import io.opentelemetry.api.GlobalOpenTelemetry
import io.opentelemetry.api.metrics.LongCounter

class TelemetryService {

    private val articlesCreated: LongCounter = GlobalOpenTelemetry.getMeter("ktor-articles")
        .counterBuilder("articles.created")
        .setDescription("Total number of articles created")
        .build()

    fun incrementArticlesCreated() {
        articlesCreated.add(1)
    }
}
